macro/derive/src/lib.rs

```diff
@@ -12,19 +15,117 @@ const SKIP_TRY_FROM_ATTR: &str = "skip_try_from";
         )]
         #[cfg_attr(feature = "ffi_import", iroha_ffi::ffi_import)]
         #item
+    })
+}
+
+#[derive(darling::FromDeriveInput, Debug)]
+#[darling(supports(enum_any))]
+struct FromVariantInput {
+    ident: syn2::Ident,
+    generics: syn2::Generics,
+    data: darling::ast::Data<SpannedValue<FromVariantVariant>, darling::util::Ignored>,
+}
+
+// FromVariant manually implemented for additional validation
+#[derive(Debug)]
+struct FromVariantVariant {
+    ident: syn2::Ident,
+    fields: darling::ast::Fields<SpannedValue<FromVariantField>>,
+}
+
+impl FromVariantVariant {
+    fn can_from_be_implemented(
+        fields: &darling::ast::Fields<SpannedValue<FromVariantField>>,
+    ) -> bool {
+        fields.style == darling::ast::Style::Tuple && fields.fields.len() == 1
+    }
+}
+
+impl darling::FromVariant for FromVariantVariant {
+    fn from_variant(variant: &syn2::Variant) -> darling::Result<Self> {
+        let ident = variant.ident.clone();
+        let fields = darling::ast::Fields::try_from(&variant.fields)?;
+        let mut accumulator = darling::error::Accumulator::default();
+
+        let can_from_be_implemented = Self::can_from_be_implemented(&fields);
+
+        for field in &fields.fields {
+            if (field.skip_from || field.skip_container) && !can_from_be_implemented {
+                accumulator.push(darling::Error::custom("#[skip_from], #[skip_try_from] and #[skip_container] attributes are only allowed for new-type enum variants (single unnamed field). The `From` traits will not be implemented for other kinds of variants").with_span(&field.span()));
+            }
+        }
+
+        for attr in &variant.attrs {
+            let span = attr.span();
+            let attr = attr.path().to_token_stream().to_string();
+            match attr.as_str() {
+                SKIP_FROM_ATTR | SKIP_TRY_FROM_ATTR | SKIP_CONTAINER => {
+                    accumulator.push(
+                        darling::Error::custom(format!(
+                            "#[{}] attribute should be applied to the field, not variant",
+                            &attr
+                        ))
+                        .with_span(&span),
+                    );
+                }
+                _ => {}
+            }
+        }
+
+        accumulator.finish()?;
+
+        Ok(Self { ident, fields })
+    }
+}
+
+// FromField manually implemented for non-standard attributes
+#[derive(Debug)]
+struct FromVariantField {
+    ty: syn2::Type,
+    skip_from: bool,
+    skip_try_from: bool,
+    skip_container: bool,
+}
+
+// implementing manually, because darling can't parse attributes that are not under some unified attr
+// It expects us to have a common attribute that will contain all the fields, like:
+// #[hello(skip_from, skip_container)]
+// The already defined macro API uses `skip_from` and `skip_container` attributes without any qualification
+// Arguably, this is also more convenient for the user (?)
```

Comment:
Imo `#[from_variant(skip_from, ...)]` is more readable since we can clearly see relation between derive macro and attribute.

Reply:
That's true. It is a bit more wordy, but the `skip_*` attrs are not used that much..